builtin/providers/aws/resource_aws_security_group.go

```diff
@@ -675,36 +682,102 @@ func resourceAwsSecurityGroupUpdateRules(
 	return nil
 }
 
+func clearSecurityGroupCache(cache *SecurityGroupsCache) {
+	mtKey := "security_groups_cache"
+	awsMutexKV.Lock(mtKey)
+	defer awsMutexKV.Unlock(mtKey)
+	cache.SecurityGroups = nil
+}
+
+func updateSecurityGroupCache(conn *ec2.EC2, cache *SecurityGroupsCache) error {
+	mtKey := "security_groups_cache"
```

Comment:
Let's make this a constant that's a unique id like we do with the [Timeout key](https://github.com/hashicorp/terraform/blob/master/helper/schema/resource_timeout.go#L12)